### PR TITLE
211 auth check in api

### PR DESCRIPTION
**Pull request by vikasrinvi**
#211 
1. passport installation 
   * command to run
      * `composer require laravel/passport`
      * `php artisan migrate`
      * `php artisan passport:install` (this will create "personal access" and "password grant" clients)
          
2. deploying passport
    * run `php artisan passport:keys`      
3. client creation (for creating any kind of client)
    * run `php artisan passport:client`
        client will be generate with id and secrete.
4. Token issue 
    * make post request in 'http://app-url/get/token' with parameter
          * grant_type
          * client_id
          * client_secret
          * scope

### CODE REVIEW

**Comment by bohraji** [comment]
@vikasrnv Please update the PR description to specify the following information clearly-

- What changes does this pull request brings in. This is for giving the reviewer to get a generic idea of what this code update will do. No need to specify the files here.
- If there are commands to be run for this update to run, please specify them clearly. Also take care that the formatting is on point.

Please update the description and assign back for review. Check out the documentation on [this](https://github.com/ColoredCow/employee-portal/pull/406) PR by vaibhav for more clarity.

File: app/Http/Controllers/Auth/LoginController.php [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
 namespace App\Http\Controllers\Auth;
 
+use App\Http\Controllers\Auth\GuzzleHttp\Client;
 use App\Http\Controllers\Controller;
 use App\User;
 use Illuminate\Foundation\Auth\AuthenticatesUsers;
 use Illuminate\Support\Facades\Auth;
 use Laravel\Socialite\Facades\Socialite;
 
+// use App\Http\Controllers\Auth\GuzzleHttp\Client;
```
Comment (bohraji) [inline app/Http/Controllers/Auth/LoginController.php]: Remove commented code

File: app/Http/Controllers/Auth/LoginController.php [hunk 2]
```diff
@@ -92,11 +95,27 @@ public function findOrCreateUser($user, $provider)
...
-            'provider_id' => $user->id
+            'provider_id' => $user->id,
+        ]);
+    }
+
+    public function GetToken()
+    {
+        $guzzle = new \GuzzleHttp\Client;
+
+        $response = $guzzle->post('http://localhost/oauth/token', [
```
Comment (bohraji) [inline app/Http/Controllers/Auth/LoginController.php]: Please remove the hard coded App URL. In production, the URL will not be `localhost`

File: app/Http/Controllers/Auth/LoginController.php [hunk 3]
```diff
@@ -92,11 +95,27 @@ public function findOrCreateUser($user, $provider)
...
+            'form_params' => [
+                'grant_type' => 'client_credentials',
+                'client_id' => '6',
```
Comment (bohraji) [inline app/Http/Controllers/Auth/LoginController.php]: Please remove hardcoded `Client ID`

Comment (bohraji) [reply app/Http/Controllers/Auth/LoginController.php]: Also, please generalise these functions.

File: app/Http/Kernel.php [hunk 4]
```diff
@@ -35,6 +36,8 @@ class Kernel extends HttpKernel
             \Illuminate\View\Middleware\ShareErrorsFromSession::class,
             \App\Http\Middleware\VerifyCsrfToken::class,
             \Illuminate\Routing\Middleware\SubstituteBindings::class,
+            // \Laravel\Passport\Middleware\CreatFreshApiToken::class,
```
Comment (bohraji) [inline app/Http/Kernel.php]: Remove commented code

File: routes/api.php [hunk 5]
```diff
@@ -11,17 +9,17 @@
 | routes are loaded by the RouteServiceProvider within a group which
 | is assigned the "api" middleware group. Enjoy building your API!
 |
-*/
+ */
 
-Route::middleware('auth:api')->get('/user', function (Request $request) {
-    return $request->user();
-});
+Route::get('/user', function (Request $request) {
```
Comment (bohraji) [inline routes/api.php]: Remove testing routes.

**Comment by bohraji** [comment]
@vikasrnv Also remove the conflicts. Also, please make sure that the routes that are now getting protected by access token are also updated with the ColoredCow website so that the workflow is not affected.
Coordinate with @rathorevaibhav on this please and update here about it.

**Comment by bohraji** [comment]
Checked this PR :

- [x] For duplication.
- [x] For performance. 
- [x] Usage of recursive methods, loops and algorithms.
- [x] For readability and understanding. 
- [x] For functions with multiple responsibilities.
- [x] For scope of refactoring. 
- [x] For coupling
- [x] For alternative approaches to the implementation. 
- [x] For how the change impacts other parts of the system.
- [x] For validations. client side and server side validations are present.

For more insights on code reviews checkout our [guidelines](https://docs.google.com/document/d/1Sa-XYJEmGwrOM5gk5zoan_Z6j90ylibmCmt331361nw/edit#).

**Comment by vikasrnv** [comment]
some changes i have made as per the suggestion 
 * I think there is no need to group the route in `api.php` with middleware
 * only `client` check require for `authorization` as client will always have `authenticated` credential if it 
    posses `access token`

**Comment by vikasrnv** [comment]
Now all the Route on `api.php` are Secure by `Oauth` 
so only client with `access token`  can make the API request

**Comment by bohraji** [comment]
@rathorevaibhav This issue is ready for deployment. There are some steps required to generate the access tokens which have been described in the PR description. Please discuss with @vikasrnv for any details.
**Also, all API routes are now being protected by `client credential grant tokens`** So we need to update the implementation on `ColoredCow website` and also the `weeklydose plugin` before merging this.

**Pull request closed without merging**